### PR TITLE
Fix camera after dying

### LINKED ISSUE #1571: [BUG] After dying the player + camera still move
## 🐛 Bug Report

After dying the player + camera still move
Out TL suggested it could be a client prediction issue

This is different from issue #1257 but they are related in camera expected behavior.

### PR DESCRIPTION
**Pull request by BertovDev**
Closes #1571 

## Motivation

After death players still can move due to client movement.

## Summary of changes

Restrict movement only when player health from Entity is > 0

## How has this been tested?

- Start a game with +1 player, force your death while walking
- When you are death the player should stop walking.

## Test additions / changes

List tests added/updated.

## Checklist
- [x] I have tested the changes locally.
- [x] I self-reviewed the changes on GitHub, line by line.
- [ ] Tests have been added/updated.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [ ] Tested in Android.

### CODE REVIEW

**Review by AngieDutra — CHANGES_REQUESTED** [review]
@tkz00, @BertovDev and I were playing. I, AN (Android build), killed them both but they both redirected to another player, also called AN (Unity desktop)

File: client/Assets/Scripts/UI/KillFeedManager.cs [hunk 1]
```diff
@@ -93,5 +99,29 @@ public void Update()
...
+
+    public ulong GetMyKillerId(){
+        return this.myKillerId;
+    }
+
+    public ulong GetcurrentTrackedPlayer(){
+        return this.currentTrackedPlayer;
+    }
+
+    public void SetcurrentTrackedPlayer(ulong newcurrentTrackedPlayer){
```
Comment (AngieDutra) [inline client/Assets/Scripts/UI/KillFeedManager.cs:124]:
```suggestion
    public void SetCurrentTrackedPlayer(ulong newCurrentTrackedPlayer){
```

File: client/Assets/Scripts/UI/KillFeedManager.cs [hunk 2]
```diff
@@ -93,5 +99,29 @@ public void Update()
...
+        this.currentTrackedPlayer = newcurrentTrackedPlayer;
```
Comment (AngieDutra) [inline client/Assets/Scripts/UI/KillFeedManager.cs:125]:
```suggestion
        this.currentTrackedPlayer = newCurrentTrackedPlayer;
```

File: client/Assets/Scripts/CustomLevelManager.cs [hunk 3]
```diff
@@ -340,26 +329,30 @@ private IEnumerator ShowDeathSplash(GameObject player)
...
+        if (saveKillerId != 0)
+        {
+            playerToFollow = Utils.GetGamePlayer(saveKillerId); 
+            StartCoroutine(WaitToChangeCamera(playerToFollow));
+        }
+        else if(Utils.GetAlivePlayers().Count() > 0 && saveKillerId == currentTrackedPlayer)
+        {
+            playerToFollow = Utils.GetAlivePlayers().ElementAt(0);
+            SetCameraToPlayer(playerToFollow.Id);
+            KillFeedManager.instance.SetcurrentTrackedPlayer(playerToFollow.Id);
```
Comment (AngieDutra) [inline client/Assets/Scripts/CustomLevelManager.cs:346]:
```suggestion
            KillFeedManager.instance.SetCurrentTrackedPlayer(playerToFollow.Id);
```

File: client/Assets/Scripts/CustomLevelManager.cs [hunk 4]
```diff
@@ -340,26 +329,30 @@ private IEnumerator ShowDeathSplash(GameObject player)
...
-    //     else
-    //     {
-    //         playerToFollow = Utils.GetAlivePlayers().ElementAt(0);
-    //         setCameraToPlayer(playerToFollow.Id);
-    //     }
-    // }
+    private void SetCameraToAlivePlayer()
+    {
+        ulong saveKillerId = KillFeedManager.instance.GetSaveKillderId();
+        ulong currentTrackedPlayer = KillFeedManager.instance.GetcurrentTrackedPlayer();
```
Comment (AngieDutra) [inline client/Assets/Scripts/CustomLevelManager.cs:335]:
```suggestion
        ulong currentTrackedPlayer = KillFeedManager.instance.GetCurrentTrackedPlayer();
```

**Review by tkz00 — APPROVED** [review]
Looks good

File: client/Assets/Scripts/UI/KillFeedManager.cs [hunk 5]
```diff
@@ -93,5 +99,29 @@ public void Update()
...
+
+    public void SetSaveKillderId(ulong newSaveKillderId){
+        this.saveKillerId = newSaveKillderId;
+    }
+
+    public ulong GetMyKillerId(){
+        return this.myKillerId;
+    }
+
+    public ulong GetcurrentTrackedPlayer(){
```
Comment (AngieDutra) [inline client/Assets/Scripts/UI/KillFeedManager.cs:120]:
```suggestion
    public ulong GetCurrentTrackedPlayer(){
```

**Review by AngieDutra — CHANGES_REQUESTED** [review]
Since this is prompt to error I would change this locally but you could also accept my changes and test this after

**Comment by BertovDev** [comment]
> Since this is prompt to error I would change this locally but you could also accept my changes and test this after

All done!

**Review by tkz00 — CHANGES_REQUESTED** [review]
During playtest this was shown to not work as expected.

**Review by AngieDutra — CHANGES_REQUESTED** [review]
Tested on Android and Unity desktop but It didn't work as expected. Sometimes it wouldn't redirect to the killer but to the first player in the list

**Review by AngieDutra — APPROVED** [review]
Awesome. Exited to restore this feature!

**Review by tkz00 — APPROVED** [review]
Looks good

**Pull request merged**